backend/tests/test_media_handler.py

```diff
@@ -0,0 +1,93 @@
+import tempfile
+import unittest
+from unittest.mock import patch, MagicMock
+from pathlib import Path
+from backend.media_handler import MediaHandler
+
+class TestMediaHandler(unittest.TestCase):
+
+    def setUp(self):
+        """Set up base directory and mock paths for testing."""
+        self.base_directory = tempfile.mkdtemp()
+        self.video_url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
+        self.config_path = "/path/to/config.json"
+        self.video_path = "/path/to/video.mp4"
+        self.input_data = {
+            "video_file_path": self.video_path,
+            "config_file_path": self.config_path
+        }
+
+    @patch("backend.media_handler.yt_dlp.YoutubeDL")
+    def test_download_media(self, mock_yt_dlp):
+        # Mock YoutubeDL instance
+        mock_ydl_instance = MagicMock()
+        
+        # Mock extract_info for download=False and download=True
+        mock_ydl_instance.extract_info.side_effect = [
+            {"title": "Test Video", "ext": "mp4"},  # For download=False
+            {"ext": "mp4"}  # For download=True
+        ]
+        
+        # Set the return value when instantiating YoutubeDL
+        mock_yt_dl_context_manager = MagicMock()
+        mock_yt_dl_context_manager.__enter__.return_value = mock_ydl_instance
+        mock_yt_dlp.return_value = mock_yt_dl_context_manager
+
+        # Test download_media method
+        result = MediaHandler.download_media(self.video_url, self.base_directory)
+        expected_file = Path(self.base_directory) / "Test_Video.mp4"
+        self.assertEqual(result, str(expected_file.resolve()))
```

Comment:
Sometimes due to rate limiting yt-dlp takes minutes to finish. Please add a timeout, e.g. 10 seconds, here for getting the info which should be rather fast irrespective of media size.

Reply:
Hey do you want to add the time out on the main code too ?

Reply:
Yes would be good